components/ui/Dialog.tsx

```diff
@@ -0,0 +1,250 @@
+        // contentMain: '',
+      },
+      [DialogVariants.Confirm]: {
+        // contentMain: '',
+      },
+      [DialogVariants.Info]: {
+        // contentMain: '',
+      },
+    },
+  },
+});
+
+// Main Dialog Component
+type DialogProps = {
+  dialog: DialogType;
+  dialogOrder: number;
+  handleOpenChange: (dialog: DialogType) => void;
+  confirmDialog: (dialog: DialogType) => void;
+  cancelDialog: (dialog: DialogType) => void;
+};
+
+const Dialog = (props: DialogProps) => {
+  const { dialog, handleOpenChange, cancelDialog, confirmDialog, dialogOrder } =
+    props;
+
+  const { title, contentMain, overlay, content, footer, close } =
+    dialogElements({
+      type: dialog.type,
+    });
+
+  return (
+    <DialogPrimitive.Root open onOpenChange={() => handleOpenChange(dialog)}>
+      <DialogPrimitive.Portal>
+        <DialogPrimitive.Overlay className={overlay()}>
+          <DialogContent className={content()} dialogOrder={dialogOrder}>
+            <DialogPrimitive.Title className={title()}>
+              <DialogIcon variant={dialog.type} />
+              <span>{dialog.title}</span>
+            </DialogPrimitive.Title>
+            <main className={contentMain()}>
+              {dialog.content}
+              <br />
+              {dialog.type === 'Error' && <DialogError {...dialog.error} />}
+            </main>
+            <DialogFooter
+              className={footer()}
+              dialog={dialog}
+              cancelDialog={cancelDialog}
+              confirmDialog={confirmDialog}
+            />
+            <DialogClose
+              className={close()}
+              disable={dialog.type === 'Prompt'}
+            />
+          </DialogContent>
+        </DialogPrimitive.Overlay>
+      </DialogPrimitive.Portal>
+    </DialogPrimitive.Root>
+  );
+};
+export default Dialog;
+
+// Dialog Content Section
+type DialogContentProps = {
+  dialogOrder: number;
+} & React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content>;
+
+const DialogContent = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Content>,
+  DialogContentProps
+>(({ children, dialogOrder, ...props }, ref) => {
+  const variants = React.useMemo(
+    () => ({
+      visible: {
+        opacity: 1,
+        translateY: '-50%',
+        translateX: '-50%',
+        scale: 1,
+        transition: { delay: dialogOrder * 0.25 },
+      },
+      hidden: {
+        opacity: 0,
+        translateY: '-50%',
+        translateX: '-50%',
+        scale: 0.8,
+      },
+    }),
+    [dialogOrder],
+  );
+
+  return (
+    <DialogPrimitive.Content ref={ref} {...props} asChild>
+      <motion.div
+        initial={'hidden'}
+        animate={'visible'}
+        exit={'hidden'}
+        variants={variants}
+      >
+        {children}
+      </motion.div>
+    </DialogPrimitive.Content>
+  );
+});
+DialogContent.displayName = DialogPrimitive.Content.displayName;
+
+// Dialog Close Button
+const DialogClose = ({
+  disable,
```

Comment:
I think its bad UX to have a close button be visible, but disabled. The button is too small for a disabled state that can be visually discerned, so having it there will make users think it just isn't working. Why not just conditionally render it?